test/utils/k8s/job.go

```diff
@@ -175,22 +176,50 @@ func AssertJob(t *testing.T, job *batchv1.Job) {
 }
 
 func GetJobLogs(ns string, kubeClient kubernetes.Interface, name string) (string, error) {
+	return GetJobsLogs(ns, kubeClient, name, false)
+}
+
+// Returns the logs of the pods related to a given job.  If 'all' is true, all runs will be
+// returned; failures to get individual logs will be reported on the output, but not cause
+// the function to fail.
+//
+// If 'all' is false, only the last run's logs will be returned.
+func GetJobsLogs(ns string, kubeClient kubernetes.Interface, name string, all bool) (string, error) {
 	pods, err := kube.GetPods(fmt.Sprintf("job=%s", name), ns, kubeClient)
 	if err != nil {
 		return "", err
 	}
-	// Consider just the latest pod
-	latestPod := pods[0]
-	for _, pod := range pods {
-		if pod.CreationTimestamp.After(latestPod.CreationTimestamp.Time) {
-			latestPod = pod
-		}
```

Comment:
This must be preserved. The pods returned by the `List` method are not sorted by creation timestamp.

Reply:
I'm sorting the list (see line 195):

	sort.Slice(pods, func(i, j int) bool {
		return pods[i].CreationTimestamp.Time.Before(pods[j].CreationTimestamp.Time)
	})